### DISCUSSION TITLE
How do I get a WFS catalog item to use the default style defined in GeoServer?

### DISCUSSION THREAD

**Discussion by lstillwe** [TerriaJS/terriajs#6422] [Q&A] [answered]
I have defined a defaultStyle for a WFS type layer in Geoserver, but is does not appear to be used in TerriaMap.
How do I get TerriaMap to use that style?

If not possible, and I need to use StyleTraits, is there anyway to use rules and filters (like sld), so I can use different colored icons based on a value defined in the point feature?

Thanks very much,
Lisa

**Comment by nf-s** [comment, marked as answer]
Hi Lisa

We have a new "Edit Style" mode in Terria - which you can use to generate style JSON

For example:
- Go to http://ci.terria.io/main/
- Load your WFS layer
- Click "Edit Style"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037496-ee76ca2c-d275-4247-be0d-69c181e71e1f.png">
- Set style to your liking
- Click "show advanced options"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037527-eb76afec-517f-4877-bd5a-dc946f982089.png">
- Click "Copy user stratum to clipboard"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037547-5471f26b-2e55-41dd-a9d9-0a3c89835807.png">
- Merge copied JSON into your catalog item JSON

**Note** this feature is brand new and experimental - so you may run into bugs/quirks

**Comment by samareshdasglobal** [comment]
Thanks for you response. is there any way to add new colormap and make that default in code?

**Reply by nf-s** [reply]
Hi @samareshdasglobal 

You can create a new discrete colour map - for example - here you can change the colour of each bin

<img width="1279" alt="image" src="https://github.com/TerriaJS/terriajs/assets/6187649/13a0cad5-66cb-403f-8e9c-4e4b868b2acc">

Currently there is no way to create a new continuous color map.

To make a colour map the default, you can use the "Copy user stratum to clipboard" function demonstrated above to copy the new colour map into your catalog configuration file
